README: Update usage instructions (#444)
NLeSC/python-template: README.md

## How to use

> There are multiple ways to use this template:

Reviewer: ```suggestion
There are multiple scenarios to use this template:
```

> 1. Creating a new package based on the template

Reviewer: ```suggestion
1. Generating a new package using template
```

> 2. Applying the template's best practices to some pre-existing code

Reviewer 2: ```suggestion
2. Applying the template to some pre-existing code
```

> 1. Creating a new package based on the template
> 2. Applying the template's best practices to some pre-existing code

Reviewer 2: ```suggestion
Option 1: Creating a new package based on the template
Option 2: Applying the template's best practices to some pre-existing code
```

> 3. Updating a package made with the template based on the latest template updates

Reviewer: ```suggestion
3. Updating a package made with the template which includes updating the answers to previous questions or syncing update from template 
```

> 3. Updating a package made with the template based on the latest template updates

Reviewer 2: ```suggestion
3. Updating a package made with the template
```

> 3. Updating a package made with the template based on the latest template updates

Reviewer 2: ```suggestion
Option 3: Updating a package made with the template based on the latest template updates
```

> In all three cases, you will need to install copier first:

Reviewer: ```suggestion
In all three scenarios, you will need to install copier first:
```

```shell

> pipx install copier

Reviewer 2: Does user also need to install pipx?

Author: I've added instructions and a note about the difference between using `pip` or `pipx`

```

> ### Option 1: Create a new package

Reviewer: ```suggestion
### Scenario 1: Generating a new package
```

> #### Step 1/2: Generate the files and directory structure

Reviewer: ```suggestion
#### Step 1/2: Create the files and directory structure
```

Once the project files have been generated, follow the steps outlined in

> [template/next_steps.md](template/next_steps.md).

Reviewer: ```suggestion
[next_steps.md](next_steps.md).
```

```

For an explanation of what's there, read on in the [project_setup.md](template/project_setup.md) file.

> ### Option 2: Apply to pre-existing code

Reviewer: ```suggestion
### Scenario 2: Applying to pre-existing code
```

To apply the template to pre-existing code, you can use the same `copier copy`
command as when creating a new package, except that you point to the folder
containing your existing code rather than a new one:

```shell
copier copy https://github.com/nlesc/python-template.git path/to/existing/code
```

This works because if `path/to/destination` already exists, `copier` will
update what is already there by either adding new files or updating
existing files. Copier will ask to overwrite any files that resulted in
conflicts. Especially if your files are already under version control, it is
recommended to answer 'yes' for all files, you will still be able to review
the changes suggested by the template.

> ### Option 3. Updating a template-made package

Reviewer: ```suggestion
### Scenario 3. Updating a template-made package
```

Copier provides the functionality for re-applying the template to a previously
created project using the `copier update` command. This has two effects:

1. Your project will be updated according to the latest version of the template
2. You can change any of your previous answers to apply these changes

>    throughout your entire project.

Reviewer: Please, never manually update answers in `.copier-answers.yml` as it will result in unexpected behavior.

Author: I've added a `> [!CAUTION]` block with this information
